devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -0,0 +1,372 @@
+	descMap.additionalAttrs.each {
+		attrData << [cluster: descMap.clusterInt, attribute: it.attrInt, value: it.value]
+	}
+	attrData.each {
+		def map = [:]
+		if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == LOCAL_TEMPERATURE) {
+			log.debug "TEMP"
+			map.name = "temperature"
+			map.value = getTemperature(it.value)
+			map.unit = temperatureScale
+		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == COOLING_SETPOINT) {
+			log.debug "COOLING SETPOINT"
+			map.name = "coolingSetpoint"
+			map.value = getTemperature(it.value)
+			map.unit = temperatureScale
+		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == HEATING_SETPOINT) {
+			log.debug "HEATING SETPOINT"
+			map.name = "heatingSetpoint"
+			map.value = getTemperature(it.value)
+			map.unit = temperatureScale
+		} else if (it.cluster == THERMOSTAT_CLUSTER && (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE)) {
+			log.debug "MODE"
+			map.name = "thermostatMode"
+			map.value = THERMOSTAT_MODE_MAP[it.value]
+		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == THERMOSTAT_RUNNING_STATE) {
+			log.debug "RUNNING STATE"
+			def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
+			map.name = "thermostatOperatingState"
+			if(binValue[0] == "1") {
+				map.value = "heating"
+			} else if(binValue[1] == "1") {
+				map.value = "cooling"
+			} else {
+				map.value = binValue[2] == "1" ? "fan only" : "idle"
+			}
+		} else if (it.cluster == FAN_CONTROL_CLUSTER && it.attribute == FAN_MODE) {
+			log.debug "FAN MODE"
+			map.name = "thermostatFanMode"
+			map.value = FAN_MODE_MAP[it.value]
+		} else if(it.cluster == zigbee.POWER_CONFIGURATION_CLUSTER && it.attribute == BATTERY_VOLTAGE) {
+			map = getBatteryPercentage(Integer.parseInt(it.value, 16))
+		}
+		if(map) {
+			result << createEvent(map)
+		}
+	}
+	return result
+}
+
+def installed() {
+	refresh()
+}
+
+def refresh() {
+	return zigbee.readAttribute(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_RUNNING_STATE) +
+			zigbee.readAttribute(FAN_CONTROL_CLUSTER, FAN_MODE) +
+			zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, BATTERY_VOLTAGE)
+}
+
+def ping() {
+	refresh()
+}
+
+def configure() {
+	def binding = zigbee.addBinding(THERMOSTAT_CLUSTER) + zigbee.addBinding(FAN_CONTROL_CLUSTER)
+	def startValues = zigbee.writeAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT, DataType.INT16, 0x07D0) +
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT, DataType.INT16, 0x0A28)
+
+	return binding + startValues + zigbee.batteryConfig() + refresh()
+}
+
+def getBatteryPercentage(rawValue) {
+	def result = [:]
+	result.name = "battery"
+	if(rawValue == 0) {
```

Comment:
if the device supports having batteries installed while it's plugged in, I'd manually change this to 100% just to make sure users aren't constantly changing batteries in a plugged-in thermostat.